### ISSUE TITLE
Deprecate default "nodejs12.x" lambda runtime

### ISSUE THREAD

**Issue by medikoo** [serverless/serverless#8921]
### Use case description

We need to switch the default to `nodejs14.x`, still it'll be breaking a change, therefore we should not do it before next major.

To pave the path for that change, first we need to deprecate `nodejs12.x` as the default.

### Proposed solution

When in AWS service configuration there's no `provider.runtime` set and there's at least one function with `functions[].handler` set and no `functions[].runtime` set, let's show the [deprecation](https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md#configuring-deprecations) stating that with next major default will be changed to `nodejs14.x` and advise user set runtime explicitly to have deprecation hidden

**Comment by endingwithali** [comment]
I'll work on this!

**Comment by medikoo** [comment]
@endingwithali thank you, great to hear that. We look forward the PR!

**Comment by endingwithali** [comment]
hey! spent some time today digging in - new to the code base so i was wondering when you said "AWS service configuration" was there a specific file in mind?

**Comment by medikoo** [comment]
> "AWS service configuration" was there a specific file in mind?

Yes, in Framework we support also other providers (although they're handled via plugin). We need to ensure to not show deprecation if user is not relying on AWS provider. Here's how to confirm on that: https://github.com/serverless/serverless/blob/769214cb2a234968fd2e76391fdcebb179cfddab/lib/plugins/aws/package/compile/events/cloudFront.js#L199

**Comment by pgrzesik** [comment]
Hello @endingwithali :wave: Are you still interested in working on this one? There's someone that expressed interest in working on this by issuing a PR (it will definitely need changes though - https://github.com/serverless/serverless/pull/8981), but before moving any further with that I wanted to confirm the status on your side. Thanks in advance :bow:

**Comment by jkklapp** [comment]
take

**Comment by pgrzesik** [comment]
Hello @jkklapp, thanks for being interested in this one :raised_hands: There's a one PR linked but the author is unresponsive, so feel free to submit a new PR that implements this change :+1:

**Comment by jkklapp** [comment]
@medikoo can we close this issue?

**Comment by pgrzesik** [comment]
Hello @jkklapp - was that addressed? I don't think it was but we'd be happy to accept a PR for this one :raised_hands:

**Comment by vaidyamilind** [comment]
Hey @pgrzesik  - we need deprecation warning after compiling all lambda? or for each lambda? previously raised pull request for each lambda. Do I need to change.

**Comment by pgrzesik** [comment]
Hello @vaidyamilind - due to long inactivity on your part, @jkklapp picked up this ticket, and looks like he's getting close to finishing it. Sorry for the inconvenience there.
